backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/trace/HUTraceEventsService.java

```diff
@@ -727,14 +721,19 @@ void createAndAddEventsForPOIssueOrReceipt(
 			else
 			{
 				builder.topLevelHuId(huId);
-				createTraceForPOIssueOrReceiptHU(builder, ppCostCollector, huRecord);
 			}
-
 			final List<I_M_HU> vhus = huAccessService.retrieveVhus(huId);
-
-			for (final I_M_HU vhu : vhus)
+			if (vhus.isEmpty())
```

Comment:
I'd rather go with something like: isCU(huId) => create the events, else retrieve the vhus and create the events for them